lib/cell/concept.rb

```diff
@@ -5,8 +5,23 @@ class Concept < Cell::ViewModel
 
     # TODO: this should be in Helper or something. this should be the only entry point from controller/view.
     class << self
-      def class_from_cell_name(name)
-        name.camelize.constantize
+      def class_from_cell_name(name, controller)
+        @cell_class ||= begin
+          supports_path = controller.supports_path?
+          routes  = controller.respond_to?(:_routes)  && controller._routes
+          helpers = controller.respond_to?(:_helpers) && controller._helpers
+
+          Class.new(name.camelize.constantize) do
```

Comment:
While that is usually true, this implementation is found here: https://github.com/rails/rails/blob/master/actionview/lib/action_view/rendering.rb#L36

This is a rails core implementation. I used it because it's in the core, which means it is tested, and because it had been able to provide a pragmatic solution.